lib/charms/observability_libs/v1/cert_handler.py

```diff
@@ -87,15 +89,164 @@ class CertHandlerEvents(ObjectEvents):
     cert_changed = EventSource(CertChanged)
 
 
+class _VaultBackend(abc.ABC):
+    def store(self, contents: Dict[str, str], clear: bool = False): ...
+
+    def get_value(self, key: str) -> Optional[str]: ...
+
+    def retrieve(self) -> Dict[str, str]: ...
+
+    def nuke(self): ...
+
+
+class _RelationVaultBackend(_VaultBackend):
+    """Relation backend for Vault.
+
+    Use it to store data in a relation databag.
+    Assumes that a single relation exists and its data is readable.
+    If not, it will raise RuntimeErrors as soon as you try to read/write.
+    It will store the data, in plaintext (json-dumped) nested under a configurable
+    key in the **unit databag** of this relation.
+    """
+
+    def __init__(self, charm: CharmBase, relation_name: str, nest_under: str = "secret-contents"):
```

Comment:
"Vault" is private to cert_handler so I imagine we could avoid exposing `nest_under` as an arg. Upgrades would be easier if we hardcode the key. For example we could set it to 
```python
self.nest_under = "lib.charms.observability_libs.v1.cert_handler::secret-contents"
```
It would still be an unenforceable convention, but everything is.

Reply:
I meant it for generality, if someone ever wants to use multiple Vaults and/or _RelationVaultBackend instances and have them backed by the same relation, they'll need to use a separate key or they'll overwrite each other

Reply:
but for now we can skip that and make a note, agreed